presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -370,4 +373,17 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                                                 project(ImmutableMap.of("NON_NULL", expression("true")),
                                                                                         node(ValuesNode.class)))))))))));
     }
+
+    @Test
+    public void testPruneCountAggregationOverScalar()
```

Comment:
@kokosing I've addressed all your comments. The thing I see is that for the `SELECT EXISTS(SELECT sum(orderkey) from orders)` query, the plan gets rewritten to a scalar value even before optimization. I believe that happens here: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java#L284.
If I understand this correctly, the rule I'm adding will not see an Exists node because of this rewrite. Please let me know if that is correct.